plugins/module_utils/s3.py

```diff
@@ -1,28 +1,24 @@
 # Copyright (c) 2018 Red Hat, Inc.
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-from ansible.module_utils.basic import to_text
+import string
 from urllib.parse import urlparse
 
-from .botocore import boto3_conn
-
 try:
-    from botocore.client import Config
-    from botocore.exceptions import BotoCoreError, ClientError
+    from hashlib import md5
+    HAS_MD5 = True
 except ImportError:
-    pass    # Handled by the calling module
+    HAS_MD5 = False
 
-HAS_MD5 = True
 try:
-    from hashlib import md5
+    import botocore
 except ImportError:
-    try:
-        from md5 import md5
```

Comment:
md5.md5 was a Python 2.x thing, no point trying to import it now we only support Python 3.6+